geode-core/src/test/java/org/apache/geode/cache/query/internal/QCompilerTest.java

```diff
@@ -0,0 +1,484 @@
+    validationHelperForCompiledJunction((CompiledJunction) result, lowerBoundKey, "abd");
+
+    s1 = "abc"
+        + new String(new char[] {(char) 255, (char) 255, (char) 255, (char) 255, (char) 255, '%'});
+    literal = new CompiledLiteral(s1);
+    compiler = new QCompiler();
+    result = compiler.createCompiledValueForLikePredicate(var, literal);
+    lowerBoundKey =
+        "abc" + new String(new char[] {(char) 255, (char) 255, (char) 255, (char) 255, (char) 255});
+    validationHelperForCompiledJunction((CompiledJunction) result, lowerBoundKey, "abd");
+
+    s1 = "%";
+    literal = new CompiledLiteral(s1);
+    compiler = new QCompiler();
+    result = compiler.createCompiledValueForLikePredicate(var, literal);
+    assertThat(result).isInstanceOf(CompiledComparison.class);
+    CompiledComparison cc = (CompiledComparison) result;
+    assertThat(cc.reflectOnOperator((CompiledValue) cc.getChildren().get(1)))
+        .isEqualTo(OQLLexerTokenTypes.TOK_GE);
+    assertThat(((CompiledLiteral) cc.getChildren().get(1))._obj).isEqualTo("");
+  }
+
+  private void validationHelperForCompiledJunction(CompiledJunction result, String lowerBoundKey,
+      String upperBoundKey) {
+    assertThat(result.getOperator()).isEqualTo(OQLLexerTokenTypes.LITERAL_and);
+    List list = result.getChildren();
+    CompiledComparison lowerBound = (CompiledComparison) list.get(0);
+    CompiledComparison upperBound = (CompiledComparison) list.get(1);
+    assertThat(lowerBound.reflectOnOperator((CompiledValue) lowerBound.getChildren().get(1)))
+        .isEqualTo(OQLLexerTokenTypes.TOK_GE);
+    assertThat(upperBound.reflectOnOperator((CompiledValue) upperBound.getChildren().get(1)))
+        .isEqualTo(OQLLexerTokenTypes.TOK_LT);
+    assertThat(((CompiledLiteral) lowerBound.getChildren().get(1))._obj).isEqualTo(lowerBoundKey);
+    assertThat(((CompiledLiteral) upperBound.getChildren().get(1))._obj).isEqualTo(upperBoundKey);
+  }
+
+  @Test
+  public void testLowestString() {
+    SortedMap<String, String> map = new TreeMap<>();
+    map.put("ab", "value");
+    map.put("z", "value");
+    map.put("v", "value");
+    SortedMap returnMap = map.tailMap(CompiledLike.LOWEST_STRING);
+    assertThat(returnMap.size()).isEqualTo(3);
+  }
+
+  @Test
+  public void testBoundaryChar() {
+    String s1 = "!";
+    String s2 = "9";
+    String s3 = "A";
+    String s4 = "[";
+    String s5 = "a";
+    String s6 = "{";
+    String s7 = new String(new char[] {(char) 255});
+    assertThat(s2.compareTo(s1) > 0).isTrue();
+    assertThat(s3.compareTo(s2) > 0).isTrue();
+    assertThat(s4.compareTo(s3) > 0).isTrue();
+    assertThat(s5.compareTo(s4) > 0).isTrue();
+    assertThat(s6.compareTo(s5) > 0).isTrue();
+    assertThat(s7.compareTo(s6) > 0).isTrue();
+  }
+
+  @Test
+  @Parameters({"MIN", "MAX", "AVG", "SUM", "COUNT"})
+  public void parsingShouldSucceedWhenAggregatesAreUsedAsPartOfTheWhereClauseWithinAnInnerSelectQuery(
```

Comment:
It will be nice to have these queries in the AggregateQueryTests (if any); most of the time when we look for supported queries, we look at associated query tests.

Reply:
👍